Branch-SDK/src/io/branch/referral/BranchViewHandler.java

```diff
@@ -138,36 +129,82 @@ public boolean shouldOverrideUrlLoading(WebView view, String url) {
+                        if (branchViewDialog_ != null) {
+                            branchViewDialog_.dismiss();
+                        }
                     }
                     return isHandled;
                 }
 
                 @Override
+                public void onPageStarted(WebView view, String url, Bitmap favicon) {
+                    super.onPageStarted(view, url, favicon);
+                }
+
+                @Override
+                public void onReceivedError(WebView view, int errorCode, String description, String failingUrl) {
+                    super.onReceivedError(view, errorCode, description, failingUrl);
+                    webViewLoadError_ = true;
+                }
+
+                @Override
                 public void onPageFinished(WebView view, String url) {
                     super.onPageFinished(view, url);
-                    layout.setVisibility(View.VISIBLE);
-                    view.setVisibility(View.VISIBLE);
-                    dialog.show();
-                    showViewWithAlphaAnimation(layout);
-                    showViewWithAlphaAnimation(view);
+                    openBranchViewDialog(branchView, callback, webView);
                 }
             });
+        }
+    }
 
-            dialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
-                @Override
-                public void onDismiss(DialogInterface dialog) {
-                    isBranchViewDialogShowing_ = false;
-                    if (callback != null) {
-                        if (isBranchViewAccepted_) {
-                            callback.onBranchViewAccepted(branchView.branchViewAction_, branchView.branchViewID_);
-                        } else {
-                            callback.onBranchViewCancelled(branchView.branchViewAction_, branchView.branchViewID_);
-                        }
-                    }
+    private void openBranchViewDialog(final BranchView branchView, final IBranchViewEvents callback, WebView webView) {
+        if (!webViewLoadError_ && Branch.getInstance() != null && Branch.getInstance().currentActivityReference_ != null) {
+            Activity currentActivity = Branch.getInstance().currentActivityReference_.get();
+            if (currentActivity != null) {
+                branchView.updateUsageCount(currentActivity.getApplicationContext(), branchView.branchViewID_);
+                parentActivityClassName_ = currentActivity.getClass().getName();
+
+                RelativeLayout layout = new RelativeLayout(currentActivity);
+                layout.setVisibility(View.GONE);
+                RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
+                layoutParams.setMargins(3, 3, 3, 3);
+
+                layout.addView(webView, layoutParams);
+                layout.setBackgroundColor(Color.parseColor("#11FEFEFE"));
```

Comment:
Can you paste in a screenshot of the webview? I'm not super comfortable with us hardcoding some of this display stuff. Must we add a margin/background?